From 231cb31697c52308a1286ba7209d64dc70ce6a02 Mon Sep 17 00:00:00 2001
From: "Joshi, Vishakha" <vishakha.joshi@intel.com>
Date: Thu, 12 Aug 2021 12:44:21 +0530
Subject: [PATCH 065/109] PWM module rmmod issue fix

PWM module rmmod issue(going to sleep) fixed by change in
clock framework API.

Signed-off-by: Joshi, Vishakha <vishakha.joshi@intel.com>
---
 drivers/pwm/pwm-keembay.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/pwm/pwm-keembay.c b/drivers/pwm/pwm-keembay.c
index 81d90cccd051..b64ba591e903 100644
--- a/drivers/pwm/pwm-keembay.c
+++ b/drivers/pwm/pwm-keembay.c
@@ -49,7 +49,11 @@ static inline struct keembay_pwm *to_keembay_pwm_dev(struct pwm_chip *chip)
 
 static void keembay_clk_unprepare(void *data)
 {
-	clk_disable_unprepare(data);
+	/* The API is updated from clk_disable_unprepare() to clk_disable() as a fix for pwm
+	 * module rmmod going to sleep.
+	 */
+
+	clk_disable(data);
 }
 
 static int keembay_clk_enable(struct device *dev, struct clk *clk)
-- 
2.25.1

